### PR TITLE
Adds support for placements

### LINKED ISSUE #11: Is there ability to provide placement name for Appodeal.show()?
As I understood, it is not possible to provide placement name as a param for Appodeal.show() method. 

Is it possible to implement it? Native libs provide such a feature.

### PR DESCRIPTION
**Pull request by nixxxon**
Fixes #11

We're new to Swift and Kotlin. Appreciate any feedback or input.

### CODE REVIEW

**Comment by vegidio** [comment]
@nixxxon Thanks for the PR! I will review it later today :)

File: ios/Classes/BannerDelegate.swift
Comment (vegidio) [inline ios/Classes/BannerDelegate.swift:11]: Aha, I noticed that I was using `AppodealBannerDelegate` before but to be honest I don't remember where I took that from.

In any case I consulted the documentation and it seems that `BannerViewDelegate` is the recommended way to track banner events, but I don't see any real advantage of one over the other. I agree to leave the implementation in the way you put here, but do you have more context why we should use `APDBannerViewDelegate` instead? I'm just curious 🙂

One more thing, it seems that we also have an alias for `APDBannerViewDelegate` called `AppodealBannerViewDelegate`. Can you use `AppodealBannerViewDelegate` instead since it's a little more descriptive and "swifty" than `APDBannerViewDelegate`? 😅

File: ios/Classes/SwiftAppodealFlutterPlugin.swift
Comment (vegidio) [inline ios/Classes/SwiftAppodealFlutterPlugin.swift:95]: As I can see here, when the user doesn't specify a placement name for the ad, it defaults the placement name to an empty string, but I would rather use the same approach that was used for Android when there's no placement name. In other words, check if the `placementName` is nil and then call the appropriate function.

Something like this:


```swift
if let placementName = args["placementName"] as? String {
    result(Appodeal.showAd(adType, forPlacement: placementName, rootViewController: rootViewController))
} else {
    result(Appodeal.showAd(adType, rootViewController: rootViewController))
}
```

My reason for this is because when we call `Appodeal.showAd` without the `forPlacement` parameter, the SDK will use the default placement defined in your Appodeal account. However, if the user doesn't define a placement name and we try to call `Appodeal.showAd` with an empty placement (`""`) like in the existing code, then the `Appodeal.showAd` will not try to use the default placement in the Appodeal account; it will actually try to find a placement with an empty name.

What do you think?

File: ios/Classes/AppodealBannerFactory.swift
Comment (vegidio) [inline ios/Classes/AppodealBannerFactory.swift:16]: I don't have much to say here, I just wanted to understand the motivation behind passing an instance of `SwiftAppodealFlutterPlugin` when we create a banner 🙂

**Review by vegidio — CHANGES_REQUESTED** [review]
@nixxxon Thank you very much for the PR. Sorry I couldn't review it earlier, but I'm away from the computer during the holiday season 🎅

In general it's looking good to me, but I put a few comments about some things that I would like to get your opinion about.

I believe some of them are little improvements and in other parts there are many things that I learned with your code, but I just want to understand it better before we press the Merge button 😊

File: ios/Classes/BannerDelegate.swift
Comment (debaserr) [reply ios/Classes/BannerDelegate.swift:11]: The reason we changed it is that we're now using `APDBannerView` instead of the old `banner()` implementation, which only works with `APDBannerViewDelegate`. Not really sure if there is really any difference though 🙂 Will change the name to be more "swiftly" 👍

File: lib/src/banner.dart
Comment (debaserr) [reply lib/src/banner.dart:15]: Good input, I think this sounds like a better way to do it as well!

File: ios/Classes/SwiftAppodealFlutterPlugin.swift
Comment (debaserr) [reply ios/Classes/SwiftAppodealFlutterPlugin.swift:95]: Nice catch, yes much better to only send the placement name if we have one 👍

File: ios/Classes/AppodealBannerFactory.swift
Comment (debaserr) [reply ios/Classes/AppodealBannerFactory.swift:16]: It's used a bit further down in this file when setting the `banner.delegate`, since `SwiftAppodealFlutterPlugin` has the `AppodealBannerViewDelegate` extension. As @nixxxon said we're a bit new to Swift so not sure if there is a better way of doing it 🙂

File: android/src/main/kotlin/io/vinicius/appodeal_flutter/AppodealBannerFactory.kt
Comment (nixxxon) [reply android/src/main/kotlin/io/vinicius/appodeal_flutter/AppodealBannerFactory.kt:23]: Totally agree! :+1:

**Review by vegidio — APPROVED** [review]

**Comment by vegidio** [comment]
@nixxxon @debaserr It's looking great and I just merged. Thanks for the changes! I will generate new package later today with your changes 🚀

**Comment by vegidio** [comment]
Version `0.5.0` uploaded to pub.dev. Thanks once again for the great contributions! 🚀

**Pull request merged**
